repos/system_upgrade/common/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py

```diff
@@ -64,17 +64,40 @@ def get_os_release(path):
                                       details={'details': str(e)})
 
 
+def check_target_major_version(curr_version, target_version):
+    required_major_version = int(curr_version.split('.')[0]) + 1
+    specified_major_version = int(target_version.split('.')[0])
+    if specified_major_version != required_major_version:
+        raise StopActorExecutionError(
+            message='Specified invalid major version of the target system',
+            details={
+                'Specified targed major version': str(specified_major_version),
+                'Required target major version': str(required_major_version),
+                'hint': (
+                    'The in-place upgrade is possible only to the next system'
+                    ' major version: {ver}. Current solution depends heavily on this'
+                    ' value and violation of this rule leads to various errors which'
+                    ' could be very confusing.'
```

Comment:
No need to give an explanation that reveals internals of leapp. Upgrading is possible only to the next major version, period.

Reply:
Actually good point. Dropping the explanation from the error msg. I will explain that in the comment.